frontend/src/app/shared/components/button/button.component.ts

```diff
@@ -1,20 +1,50 @@
-import { Component, EventEmitter, Input, Output } from '@angular/core';
+import { Component, EventEmitter, Input, OnChanges, OnInit, Output } from '@angular/core';
 
 @Component({
   selector: 'app-button',
   templateUrl: './button.component.html',
   styleUrls: ['./button.component.sass']
 })
-export class ButtonComponent {
-
+export class ButtonComponent implements OnChanges{
   @Input() text = '';
 
   @Input() width = 'auto';
   
   @Input() height = 'auto';
 
+  @Input() padding = "10px 20px";
+
+  @Input() fontSize = '16px';
+
+  @Input() variant = "outline-primary";
+
+  @Input() isDisabled = false;
+
   @Output() buttonOnClick: EventEmitter<void> = new EventEmitter<void>();
 
+  btnClass = "";
```

Comment:
You can use a single [ngClass] binding instead of a separate btnClass property, for example:
```
[ngClass]="{
  'outline-primary': variant === 'outline-primary',
  'filled': variant === 'filled'
}
```